app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java

```diff
@@ -471,20 +472,33 @@ public void onFabClick(boolean needsSearch) {
             for (String id : optionSetIds) {
                 queryData.remove(id);
             }
-            if (selectedProgram != null && selectedProgram.displayFrontPageList()) {
-                if (queryData.size() < selectedProgram.minAttributesRequiredToSearch()) {
+
+            if(compliesWithMinAttributesToSearch()) {
+                view.setFabIcon(false);
+                queryProcessor.onNext(queryData);
+            } else {
+                if (selectedProgram.displayFrontPageList()) {
                     showList = false;
-                    view.setFabIcon(true);
-                    queryProcessor.onNext(new HashMap<>());
-                } else {
-                    showList = true;
-                    view.setFabIcon(false);
-                    queryProcessor.onNext(queryData);
                 }
+                view.setFabIcon(true);
+                queryProcessor.onNext(new HashMap<>());
+            }
+        }
+    }
+
+    private boolean compliesWithMinAttributesToSearch() {
+        if(selectedProgram != null) {
+            if(selectedProgram.displayFrontPageList() && queryData.size() < selectedProgram.minAttributesRequiredToSearch()){
+                return false;
+            } else if (!selectedProgram.displayFrontPageList() && queryData.size() < selectedProgram.minAttributesRequiredToSearch()) {
+                return false;
+            } else if (!selectedProgram.displayFrontPageList() && queryData.size() == 0 && selectedProgram.minAttributesRequiredToSearch() == 0) {
+                return false;
```

Comment:
do you think we could extract this and perform an unit test?

Reply:
This is already handled in 2.1.0 in a different way. I think it is not needed to add now a unit test that won't be used in the latest version